Importing the necessary libraries:

In [ ]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

### **DATA UNDERSTANDING:**

In [ ]:
df=pd.read_csv("smokin2_data.csv",header= None)

In [ ]:
df.head()

0    1           2           3          4               5   \
0  ID  age  height(cm)  weight(kg)  waist(cm)  eyesight(left)   
1   0   40         155          60       81.3             1.2   
2   1   40         160          60         81             0.8   
3   2   55         170          60         80             0.8   
4   3   40         165          70         88             1.5   

                6              7               8         9   ...          16  \
0  eyesight(right)  hearing(left)  hearing(right)  systolic  ...  hemoglobin   
1                1              1               1       114  ...        12.9   
2              0.6              1               1       119  ...        12.7   
3              0.8              1               1       138  ...        15.8   
4              1.5              1               1       100  ...        14.7   

              17                18   19   20   21    22             23  \
0  Urine protein  serum creatinine  AST  ALT  Gtp  oral  dental caries   
1              1               0.7   18   19   27     1              0   
2              1               0.6   22   19   18     1              0   
3              1                 1   21   16   22     1              0   
4              1                 1   19   26   18     1              0   

       24       25  
0  tartar  smoking  
1       1        0  
2       1        0  
3       0        1  
4       1        0  

[5 rows x 26 columns]

The top 5 rows are printed, which displays the 4 data rows and the header row as data row, as we have mentioned header as none during reading csv file

In [ ]:
df.describe()

0    1    2    3    4    5    6    7    8    9   ...    16   17   18  \
count   500  500  500  500  500  500  500  500  500  500  ...   500  500  500   
unique  500   14   12   15  145   15   15    3    3   67  ...    73    5   12   
top      ID   40  170   70   85  1.2  1.2    1    1  130  ...  15.8    1  0.9   
freq      1  138   98   77   23  116  112  486  489   33  ...    19  469   94   

         19   20   21   22   23   24   25  
count   500  500  500  500  500  500  500  
unique   47   69  106    2    3    3    3  
top      24   19   15    1    0    1    0  
freq     41   30   26  499  389  287  294  

[4 rows x 26 columns]

In [ ]:
df.shape

(500, 26)

Totally, our data set contains 500 rows and 26 columns

In [ ]:
print(df)

      0    1           2           3          4               5   \
0     ID  age  height(cm)  weight(kg)  waist(cm)  eyesight(left)   
1      0   40         155          60       81.3             1.2   
2      1   40         160          60         81             0.8   
3      2   55         170          60         80             0.8   
4      3   40         165          70         88             1.5   
..   ...  ...         ...         ...        ...             ...   
495  618   70         170          55         75             1.2   
496  619   40         170          65         82               1   
497  620   50         155          65         78             0.8   
498  621   35         170          90         95             0.9   
499  622   45         155          55       74.5               1   

                  6              7               8         9   ...  \
0    eyesight(right)  hearing(left)  hearing(right)  systolic  ...   
1                  1              1        

In [ ]:
new_df=df.drop(0)

Now, we have dropped the 0th row, that is our header row so that now our dataset contains only the numeric values

In [ ]:
new_df.head()

0   1    2   3     4    5    6  7  8    9   ...    16 17   18  19  20  21  \
1  0  40  155  60  81.3  1.2    1  1  1  114  ...  12.9  1  0.7  18  19  27   
2  1  40  160  60    81  0.8  0.6  1  1  119  ...  12.7  1  0.6  22  19  18   
3  2  55  170  60    80  0.8  0.8  1  1  138  ...  15.8  1    1  21  16  22   
4  3  40  165  70    88  1.5  1.5  1  1  100  ...  14.7  1    1  19  26  18   
5  4  40  155  60    86    1    1  1  1  120  ...  12.5  1  0.6  16  14  22   

  22 23 24 25  
1  1  0  1  0  
2  1  0  1  0  
3  1  0  0  1  
4  1  0  1  0  
5  1  0  0  0  

[5 rows x 26 columns]

In [ ]:
new_df[25].value_counts()

0    294
1    205
Name: 25, dtype: int64

In our dataset 25th column specify whether a person has smoking symptoms or not, Hence when we print the value count, it displays as 294 records with smoking symptoms and 205 records with no smoking symptoms

In [ ]:
new_df.groupby(25).mean()

<ipython-input-65-d023727f41bf>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  new_df.groupby(25).mean()


0    1    2    3              7              8    9    10   11   12   13  \
25                                                                              
0   inf  inf  inf  inf  3.779289e+290  3.779289e+290  inf  inf  inf  inf  inf   
1   inf  inf  inf  inf  5.420054e+201  5.420054e+201  inf  inf  inf  inf  inf   

     14   15             17   19   20   21             22             23  \
25                                                                         
0   inf  inf  3.779289e+290  inf  inf  inf  3.779289e+290  3.401361e+279   
1   inf  inf  5.420055e+201  inf  inf  inf  5.420054e+201  4.878537e+200   

               24  
25                 
0   3.775884e+290  
1   4.926829e+200

In [ ]:
x=new_df.drop(columns=25,axis=0)
y=new_df[25]
print(x)
print(y)

      0   1    2   3     4    5    6  7  8    9   ...   15    16 17   18  19  \
1      0  40  155  60  81.3  1.2    1  1  1  114  ...  126  12.9  1  0.7  18   
2      1  40  160  60    81  0.8  0.6  1  1  119  ...  127  12.7  1  0.6  22   
3      2  55  170  60    80  0.8  0.8  1  1  138  ...  151  15.8  1    1  21   
4      3  40  165  70    88  1.5  1.5  1  1  100  ...  226  14.7  1    1  19   
5      4  40  155  60    86    1    1  1  1  120  ...  107  12.5  1  0.6  16   
..   ...  ..  ...  ..   ...  ...  ... .. ..  ...  ...  ...   ... ..  ...  ..   
495  618  70  170  55    75  1.2  0.9  1  1  124  ...  120  17.3  1  0.9  24   
496  619  40  170  65    82    1  1.5  1  1  121  ...  123    15  1    1  48   
497  620  50  155  65    78  0.8  0.8  1  1  128  ...  161  13.7  1  0.6  25   
498  621  35  170  90    95  0.9  0.9  1  1  135  ...  160  16.7  1  0.9  27   
499  622  45  155  55  74.5    1  0.7  1  1  108  ...  137  14.5  1  0.7  43   

     20   21 22 23 24  
1    19   27  1

x has been assigned with all the data required to find the result and the y has been assigned with the result

### **TRAINING AND TESTING DATA:**

In [ ]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.1,stratify=y,random_state=1)

Now we split the data into test data and train data, and store it in corresponding varaiable, so that using the test data our model will be trained to produce result with some accuracy score, and the we can validate using
test data.

In [ ]:
print(x.shape,x_train.shape,x_test.shape)

(499, 25) (449, 25) (50, 25)


In [ ]:
model=LogisticRegression()

Since we need our model to predict the target outcome which is binary in nature (smoking or not smoking) we train our model with logistic regression



### **BUILDING MODEL:**

In [ ]:
model.fit(x_train,y_train)

/usr/local/lib/python3.10/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

Now we include the training data to our model to train the model to predict the outcome

In [ ]:
x_train_prediction = model.predict(x_train)
training_data_accuracy = accuracy_score(x_train_prediction,y_train)

In [ ]:
print("Accuracy: ",training_data_accuracy)

Accuracy:  0.7438752783964365


we can find the accuracy of our model prediction using accuracy score function, which is 74 % for our case

In [ ]:
x_test_prediction = model.predict(x_test)
testing_data_accuracy = accuracy_score(x_test_prediction,y_test)

In [ ]:
print("Accuracy: ",testing_data_accuracy)

Accuracy:  0.68


Accuracy for test prediction is 68%, which somewhat ok, as building a model that predicts with 100% accuracy is impractical

### **EVALUATING OUR MODEL:**

In [ ]:
input_data = (30,35,170,65,84.6,1,1,1,1,127,72,87,179,76,52,112,16,1,0.8,16,15,12,1,0,0,)
input_arr = np.asarray(input_data)
input_arr_reshaped = input_arr.reshape(1,-1)
prediction = model.predict(input_arr_reshaped)
if prediction[0]=='1':
  print("Has a Symptoms of smoking")
else:
  print("Has no symptoms of smoking")

Has no symptoms of smoking


Now a sample input is given to pour model which is first converted to array and then reshaped, then finally predicts the result